lib/rbnacl/group_elements/curve25519.rb

```diff
@@ -61,7 +61,9 @@ def mult(integer)
         Util.check_length(integer, SCALARBYTES, "integer")
 
         result = Util.zeros(SCALARBYTES)
-        self.class.scalarmult_curve25519(result, integer, @point)
+        if self.class.scalarmult_curve25519(result, integer, @point) != 0
+          raise CryptoError "Invalid curve25519 result"
```

Comment:
This wants to be written as `unless self.class.scalarmult_curve25519(result, integer, @point)`

The `sodium_function` wrapper that exposes the C API to ruby converts 0 returns from libsodium into `true` and all others into `false`.

Agreed on the need for a test to test this condition fails as expected though.